### PR TITLE
Remove Interop.User32 class

### PR DESCRIPTION
**Pull request by JeremyKuhne**
This change fully removes the Interop.User32 class. This does this by:

- Removing WM
- Moving GetDCScope and RegisteredMessage to Windows.Win32

Using an enum had the advantage of seeing relatively easily what a given message was (presuming it was a standard WM_ message). To help a bit with this I've created a MessageId struct that allows seeing the message description in the debugger.

One nice benefit of this change is that we now have links to the docs for the given message defines, which will help development a lot.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/9331)

### CODE REVIEW

**Comment by JeremyKuhne** [comment]
Note that the test fix was for an existing issue that I saw here.

File: src/System.Windows.Forms.Primitives/src/System/Windows/Forms/Internals/MessageDecoder.cs [hunk 1]
```diff
@@ -12,322 +10,23 @@ namespace System.Windows.Forms;
...
-
-    private static string ToString(IntPtr hWnd, User32.WM msg, WPARAM wparam, LPARAM lparam, LRESULT result)
+    public static string ToString(Message message) => ToString(
+        message.HWnd,
+        message.MsgInternal,
+        message.WParamInternal,
+        message.LParamInternal,
+        message.ResultInternal);
+
+    private static string ToString(IntPtr hWnd, MessageId msg, WPARAM wparam, LPARAM lparam, LRESULT result)
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Primitives/src/System/Windows/Forms/Internals/MessageDecoder.cs:20]: Consider replacing `msg` with a full word

File: src/System.Windows.Forms.Primitives/src/Windows/Win32/MessageId.cs [hunk 2]
```diff
@@ -0,0 +1,336 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Windows.Win32;
+
+/// <summary>
+///  Simple internal wrapper that enables showing the message identifier string in the debugger.
+/// </summary>
+internal readonly struct MessageId
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Primitives/src/Windows/Win32/MessageId.cs:10]: I read your PR description as if you added a `DebuggerDisplay` attribute. Is `ToString` called by default and it's text shown in the locals window?

**Review by Tanya-Solyanik — APPROVED** [review]

File: src/System.Windows.Forms.Primitives/src/Windows/Win32/MessageId.cs (hunk 2 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms.Primitives/src/Windows/Win32/MessageId.cs:10]: ToString is the default

File: src/System.Windows.Forms.Primitives/src/System/Windows/Forms/Internals/MessageDecoder.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms.Primitives/src/System/Windows/Forms/Internals/MessageDecoder.cs:20]: I'll iterate on this in the next PR.

**Pull request merged**